### PR TITLE
JetBrains: Get rid of IntelliJ warnings

### PR DESCRIPTION
**Pull request by vdavid**
The warnings bothered me because I like it when everything that's yellow or red is something that we should actually get rid of. There is still the `AppUIExecutor.onUiThread().execute()` call ([here](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@dv/jetbrains-fix-warnings/-/blob/client/jetbrains/src/main/java/com/sourcegraph/action/OpenSearchAction.java?L17%3A9-17%3A43=&utm_product_name=IntelliJ+IDEA&utm_product_version=IntelliJ+IDEA)) that I haven't been able to fix, but all other non-generated files are warning-free now in the `jetbrains` folder.

## Test plan

Run the code and make sure it runs as intended.
I went through the main functionality and it seems to work: https://www.loom.com/share/efcfaae8b1844c87ab29e86c114ad4e2

## App preview:

- [Web](https://sg-web-dv-jetbrains-fix-warnings.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-winmczqurd.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/jetbrains/src/main/java/com/sourcegraph/scheme/SchemeHandler.java [hunk 1]
```diff
@@ -82,16 +82,18 @@ public synchronized boolean readResponse(
     }
 
     private boolean loadContent(String resName) {
-        InputStream inStream = getClass().getResourceAsStream(resName);
-        if (inStream != null) {
-            try {
+        try (
+            InputStream inStream = getClass().getResourceAsStream(resName)
+        ) {
```
Comment (philipp-spiess) [inline client/jetbrains/src/main/java/com/sourcegraph/scheme/SchemeHandler.java:87]: what is this syntax, I've never seen that before 🙈

File: client/jetbrains/src/main/java/com/sourcegraph/ui/SourcegraphWindow.java [hunk 2]
```diff
@@ -75,29 +74,31 @@ public SourcegraphWindow(Project project) {
...
+            .setModalContext(false)
+            .setRequestFocus(true)
+            .setFocusable(true)
+            .setMovable(true)
+            .setBelongsToGlobalPopupStack(true)
+            .setCancelOnOtherWindowOpen(true)
+            .setCancelKeyEnabled(true)
+            .setNormalWindowLevel(true)
+            .createPopup();
     }
```
[the remark this answers is not in the document]
Comment (vdavid) [reply client/jetbrains/src/main/java/com/sourcegraph/ui/SourcegraphWindow.java:102]: I think it's just a coincidence. I've extracted `createPopup` intentionally from `showPopup` :) I did it in a separate commit, here: https://github.com/sourcegraph/sourcegraph/pull/34575/commits/ca1430e60bcfedb369b5f62dfbc7a16033afb51d

File: client/jetbrains/src/main/java/com/sourcegraph/scheme/SchemeHandler.java (hunk 1 above)
Comment (vdavid) [reply client/jetbrains/src/main/java/com/sourcegraph/scheme/SchemeHandler.java:87]: Me neither! It's called `try-with-resources` and I've just learned it from here: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html
It makes sure that there are no memory leaks around destroying resources.

**Comment by vdavid** [comment]
@philipp-spiess I think most of this stuff cannot be automated. There was this `try-with-resources` thing for example, which the IDE didn't give any suggestions to resolve, just some vague clue on where to look. Handling nulls were another source of warnings, and these needed manual decisions, too. I suggest we rely on IntelliJ for the coming month and make it a habit not to commit any warnings. Later on, we can look into the tooling. What do you think?

File: client/jetbrains/src/main/java/com/sourcegraph/ui/SourcegraphWindow.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (philipp-spiess) [reply client/jetbrains/src/main/java/com/sourcegraph/ui/SourcegraphWindow.java:102]: Oh gotcha. Let's make sure it's private then!

**Comment by philipp-spiess** [comment]
> @philipp-spiess I think most of this stuff cannot be automated. There was this `try-with-resources` thing for example, which the IDE didn't give any suggestions to resolve, just some vague clue on where to look. Handling nulls were another source of warnings, and these needed manual decisions, too. I suggest we rely on IntelliJ for the coming month and make it a habit not to commit any warnings. Later on, we can look into the tooling. What do you think?

Sounds reasonable, thanks for the context!

File: client/jetbrains/src/main/java/com/sourcegraph/ui/SourcegraphWindow.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (vdavid) [reply client/jetbrains/src/main/java/com/sourcegraph/ui/SourcegraphWindow.java:102]: Oh, nice catch, I've done that.

**Pull request merged**